### PR TITLE
lnrpc: Add payment hash to SendResponse

### PR DESCRIPTION
**Pull request by gitlikeagirl**
When SendPayment grpc call fails, there is currently no way of associating the failure with a payment. 
Payment hash is already extracted by extractPaymentIntent, so it can just be added to SendResponse.
Did not add hash to successful sends, since they have the preimage anyway.

### CODE REVIEW

**Comment by gitlikeagirl** [comment]
ah, thanks @ccdle12 🙈 
any idea what I can do about the timeout in `TestLightningWallet/btcwallet/neutrino:reorg_wallet_balance` (14.68s)` ? Can't see my change affecting the test.

**Comment by wpaulino** [comment]
@gitlikeagirl that is a known issue with our integration tests, I wouldn't worry about it for now.

**Comment by joostjager** [comment]
There is one more stream.send. The success case. It does contain the preimage, so the hash can be computed. For consistency I'd think it is better to always supply the hash, as it is the main key to identify a payment.

**Comment by wpaulino** [comment]
Needs a rebase!

**Comment by halseth** [comment]
LGTM after rebase!

**Comment by Roasbeef** [comment]
Ready to land after a rebase, will have to synchronize with any other PRs that also modify the protos.

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -730,6 +730,7 @@ message SendResponse {
     string payment_error = 1 [json_name = "payment_error"];
     bytes payment_preimage = 2 [json_name = "payment_preimage"];
     Route payment_route = 3 [json_name = "payment_route"];
+    bytes payment_hash = 4 [json_name ="payment_hash"];
```
Comment (halseth) [inline lnrpc/rpc.proto]: missing space after `=`

**Comment by gitlikeagirl** [comment]
Space added and rebase done. Just not sure how to address the merge conflict in rpc.pb.go, is it ok to leave as is?

**Comment by halseth** [comment]
To resolve the merge conflict, checkout the master version: `git checkout master lnrcp/rpc.pb.go` and regen the protos: `make rpc`.

**Pull request merged**